### PR TITLE
rename the digest provider files to avoid any name clashes with other…

### PR DESCRIPTION
**Pull request by slontis**
… folders

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
@jon-oracle

File: providers/default/digests/build.info [hunk 1]
```diff
@@ -1,17 +1,17 @@
 SOURCE[../../../libcrypto]=\
-        null.c
+        null_prov.c
```
Comment (levitte) [inline providers/default/digests/build.info]: Oh, right, I noticed that one yesterday...  why does it exist at all?

Note that this isn't meant to impact this PR, but is rather a heads up for another cleanup PR

**Comment by levitte** [comment]
I'd like to know what it was that gave this away?  I might encounter similar issues with #8877, so if there's some automated way we can use to detect this sort of mistake, that would be great.

File: providers/default/digests/build.info (hunk 1 above)
Comment (slontis) [reply providers/default/digests/build.info]: this is equivalent to EVP_MD null_md. I dont know if it is needed or not.

**Comment by slontis** [comment]
> here's some automated way we can use to detect this sort of mistake, that would be great.
We were using
ar -x to extract all the .o files into a folder.
Joining them back into a shared object then has missing symbols. 
Maybe the -N option will give you something..

**Comment by jon-oracle** [comment]
Here's a script I created to check for duplicate objects in the static library:

```
ARCHIVE=<path-to-.a-file>

ar -t $ARCHIVE | grep "\.o" | sort > ar_objs

rm -rf tmp_ar
mkdir tmp_ar
cd tmp_ar
  ar x ../$ARCHIVE
  ls -1 | sort > ../ar_dir
cd ../

diff ar_objs ar_dir
```

**Pull request closed without merging**